scripts/common.scala

```diff
@@ -145,53 +147,55 @@ object Progress:
   def report(amount: Int, label: String)(using progress: Progress): Unit =
     progress.increment(amount, lbl = label)
 
-  def failure(error: String)(using progress: Progress): Unit =
+  def fail(error: String)(using progress: Progress): Unit =
     progress.fail(error)
 
-  def summary(using progress: Progress): Unit =
+  def end(using progress: Progress): Unit =
     progress.end()
 
 def withProgress[A](title: String, total: Int)(f: Progress ?=> A): A =
-  val counter = new AtomicInteger(0)
-  val failed  = collection.concurrent.TrieMap.empty[String, String]
-  val first   = LocalTime.now()
-
-  def elapsed(from: Temporal, to: Temporal = LocalTime.now()): String = {
-    Duration.between(from, to).toSeconds match
-      case s if s < 60 => s"$s seconds"
-      case s           => s"${s / 60} minutes"
-  }
+  val counter   = new AtomicInteger(0)
+  val failed    = collection.concurrent.TrieMap.empty[String, String]
+  val succeeded = collection.concurrent.TrieMap.empty[String, String]
+  val first     = LocalTime.now()
+
+  def elapsed(from: Temporal, to: Temporal = LocalTime.now()): String =
+    Duration.fromNanos(java.time.Duration.between(from, to).toNanos) match
+      case d if d.toHours > 0   => f"${d.toHours} h ${d.toMinutes % 60} m ${d.toSeconds % 60} s"
+      case d if d.toMinutes > 0 => f"${d.toMinutes} m ${d.toSeconds % 60} s"
+      case d if d.toSeconds > 0 => f"${d.toSeconds} s ${d.toMillis % 1000} ms"
+      case d if d.toMillis > 0  => f"${d.toMillis} ms ${d.toMicros % 1000} µs"
+      case d                    => f"${d.toMicros} µs"
```

Comment:
why f interpolator? it's not being used, s interpolator would be enough here: https://docs.scala-lang.org/scala3/book/string-interpolation.html#the-f-interpolator-f-strings

Reply:
sorry, copilot error 😅